outputs 'env activate' verbose message on stderr

deronnax opened the pull request:
# Pull Request Check List

Resolves: #10351 

<!-- This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://python-poetry.org/docs/contributing/) at least once, it will save you unnecessary review cycles! -->

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

I don't think it needs a documentation change (this corner case wasn't documented in the first place).

<!-- If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing! -->

## Summary by Sourcery

Modify Poetry's run command to output the virtualenv activation message to stderr instead of stdout when in verbose mode

Bug Fixes:
- Ensure that the virtualenv activation message is printed to stderr when running in verbose mode

Tests:
- Updated test case to verify the virtualenv activation message is now output to stderr

radoering reviewed (changes requested):
Can you add a test in [test_activate.py](https://github.com/python-poetry/poetry/blob/main/tests/console/commands/env/test_activate.py) that checks that the output only contains the desired line and not any additional information in verbose mode?

deronnax commented:
I added the verbose mode in the existing tests, it seems better to me than adding an extra test just for this.
Also, I took the liberty to change this line:
`line = tester.io.fetch_output().rstrip("\n")`
to this
`    line = tester.io.fetch_output().split("\n")[0]`
Because to me it was much clearer on the intent (we don't want all output'd lines split by newlines and keep only the first as we could be believe, we actually want everything output'd, except the newline).

radoering commented:
> Also, I took the liberty to change this line:

Agreed.

> I added the verbose mode in the existing tests, it seems better to me than adding an extra test just for this.

Strictly speaking, we cannot be sure that it still works without verbose output if we just test the verbose mode. Thus, I think it makes sense to test both (parametrized). However, even more important, we have to use an `ApplicationTester` instead of a `CommandTester`. Otherwise, the offending line will not show up (without the fix). Thereby, we also ensure to get the most output. I just added an additional test due to the slightly different setup and to keep test concerns separate.
